src/j1client.d.ts

```diff
@@ -26,12 +26,12 @@ declare module "@jupiterone/jupiterone-client-nodejs" {
     properties: object;
   }
 
+  export type QueryOrDataResult = QueryResult[] | Object[];
+
   export default class JupiterOneClient {
     constructor(options: JupiterOneClientOptions);
-
     init(rules?: boolean): Promise<void>;
-
-    queryV1(j1ql: string): Promise<QueryResult[]>;
+    queryV1(j1ql: string): Promise<QueryOrDataResult>;
```

Comment:
Forgive my ignorance, but how are results not `QueryResult`?

Reply:
The results will differ based on the type of query. You'll get a different result if you `return tree`, and different again if you do something like `Find Person as p return p.username`.

Reply:
This can be worked around in the client-side by explicit use of `any`, but it is a little confusing when the type exported by this project conflicts with the shape of the data returned in practice.

Reply:
I don't know enough about this project, but it would be cool to have `type QueryResult = OneKindOfResult | OtherKindOfResult`, or to make the function that returns result support a generic type declaration, so that the code calling the query can declare the expected type of the response.